server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -168,6 +199,27 @@ SearchResponse transformResponse(SearchRequest request, SearchResponse response)
         0,
         Collections.emptyList(),
         Collections.emptyList(),
+        Collections.emptyList(),
         null
     );
+
+    <Result extends SearchPhaseResult> SearchPhaseResults<Result> runSearchPhaseTransformer(
+        SearchPhaseResults<Result> searchPhaseResult,
+        SearchPhaseContext context,
+        String currentPhase,
+        String nextPhase
+    ) throws SearchPipelineProcessingException {
+
+        try {
+            for (SearchPhaseResultsProcessor searchPhaseResultsProcessor : searchPhaseResultsProcessors) {
+                if (currentPhase.equals(searchPhaseResultsProcessor.getBeforePhase().getName())
+                    && nextPhase.equals(searchPhaseResultsProcessor.getAfterPhase().getName())) {
+                    searchPhaseResult = searchPhaseResultsProcessor.process(searchPhaseResult, context);
```

Comment:
Reassigning the function arguments probably should be avoided, please introduce the local variable instead

Reply:
Let me know if the reply here can fix this comment: https://github.com/opensearch-project/OpenSearch/pull/7283#discussion_r1245529685

Reply:
:clap: it does, thanks @navneet1v